### ISSUE TITLE
AtomicRex jobs break when structure identifier in training data contains '/'

### ISSUE THREAD

**Issue by pmrv** [pyiron/pyiron_contrib#423]
Caused by missing input sanitation.

Error message:
```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Input In [56], in <module>
----> 1 job.run()

File /srv/conda/envs/notebook/lib/python3.8/site-packages/pyiron_base/generic/util.py:213, in Deprecator.__deprecate_argument.<locals>.decorated(*args, **kwargs)
    203     if kw in self.arguments:
    204         warnings.warn(
    205             message_format.format(
    206                 "{}.{}({}={})".format(
   (...)
    211             stacklevel=2,
    212         )
--> 213 return function(*args, **kwargs)

File /srv/conda/envs/notebook/lib/python3.8/site-packages/pyiron_base/job/generic.py:704, in GenericJob.run(self, delete_existing_job, repair, debug, run_mode, run_again)
    702     self._run_if_repair()
    703 elif status == "initialized":
--> 704     self._run_if_new(debug=debug)
    705 elif status == "created":
    706     self._run_if_created()

File /srv/conda/envs/notebook/lib/python3.8/site-packages/pyiron_base/job/generic.py:1422, in GenericJob._run_if_new(self, debug)
   1420     print("job exists already and therefore was not created!")
   1421 else:
-> 1422     self.save()
   1423     self.run()

File /srv/conda/envs/notebook/lib/python3.8/site-packages/pyiron_base/job/generic.py:1273, in GenericJob.save(self)
   1271 if self._check_if_input_should_be_written():
   1272     self.project_hdf5.create_working_directory()
-> 1273     self.write_input()
   1274     _copy_restart_files(job=self)
   1275 self.status.created = True

File /srv/conda/envs/notebook/lib/python3.8/site-packages/pyiron_contrib/atomistics/atomicrex/base.py:221, in AtomicrexBase.write_input(self, directory)
    219 self.input._write_xml_file(directory=directory, job=self)
    220 self.potential.write_xml_file(directory=directory)
--> 221 self.structures.write_xml_file(directory=directory)

File /srv/conda/envs/notebook/lib/python3.8/site-packages/pyiron_contrib/atomistics/atomicrex/structure_list.py:415, in ARStructureContainer.write_xml_file(self, directory, name)
    411         forces = self.fit_properties["atomic-forces"]._per_element_arrays[
    412             "target_val"
    413         ][vec_start:vec_end]
    414         if not self._structures._per_chunk_arrays["predefined"][i]:
--> 415             write_modified_poscar(
    416                 identifier=self._structures.identifier[i],
    417                 forces=forces,
    418                 positions=self._structures.positions[vec_start:vec_end],
    419                 symbols=self._structures.symbols[vec_start:vec_end],
    420                 cell=self._structures.cell[i],
    421                 directory=directory,
    422             )
    423         # Maybe implement a check for forces when setting up a predefined structure?
    424         # else:
    425         #    if not np.all(np.isnan(forces)):
    426 
    427 # write xml
    428 for i in range(self._structures.num_chunks):

File /srv/conda/envs/notebook/lib/python3.8/site-packages/pyiron_contrib/atomistics/atomicrex/structure_list.py:665, in write_modified_poscar(identifier, forces, directory, structure, positions, cell, symbols)
    652 """
    653 Internal function. Writes a pyiron structure
    654 and corresponding forces in a modified POSCAR file.
   (...)
    662     directory (str): Working directory.
    663 """
    664 filename = posixpath.join(directory, f"POSCAR_{identifier}")
--> 665 with open(filename, "w") as f:
    666     # Elements as system name
    667     # Also check if the symbols are disordered
    668     # to prevent errors resulting from poscar format
    669     # not having indices or element for each atom.
    670     elements = []
    671     counter = 0

FileNotFoundError: [Errno 2] No such file or directory: '/home/jovyan/workshop_preparation/potentials/01-EAM/FitAl_hdf5/FitAl/POSCAR_/home/users/lysogy36'
```
(example path is from the workshop, where the identifiers started with /user/users/lysogy.....

This is not critical for the workshop.

**Comment by Leimeroth** [comment]
I implemented a check in the add_structure method in #443, but this won't help if the add_training_data() method is used.

Since "/" are used for paths I would suggest to directly check for that in the StructureContainer or FlattenedStorage instead of the atomicrex class?

**Comment by pmrv** [comment]
Structure identifiers in general `StructureContainers` make no promise to be valid file paths and don't need to (e.g. `Mlip` doesn't write structures to separate files, but keeps the name only in a text field inside a file).  So I don't want to fix this problem at this level.  Instead I think the easiest fix is to `identifier.replace('/', '_slash_')` or some such when you write the files and the inverse when reading them again.
